lib/system.nim

```diff
@@ -2401,42 +2416,44 @@ proc `$`*[T: tuple|object](x: T): string =
       result.add("...")
   result.add(")")
 
-proc collectionToString[T: set | seq](x: T, b, e: string): string =
-  when x is seq:
-    if x.isNil: return "nil"
-  result = b
+proc collectionToString[T](x: T, prefix, separator, suffix: string): string =
+  result = prefix
   var firstElement = true
   for value in items(x):
-    if not firstElement: result.add(", ")
+    if firstElement:
+      firstElement = false
+    else:
+      result.add(separator)
```

Comment:
Technically that is true. the separator is always ``","``  where this function is used. The reason I did add the separator as a parameter is, I can imagine very likely that this procedure might become exported from system at some point. and when it does get exported, I would really like it to have the separator as a parameter. It would be the equivalent of [``mkString`` in Scala](http://www.scala-lang.org/api/2.12.2/scala/collection/Seq.html#mkString(start:String,sep:String,end:String):String) and the additional parameter doesn't really cost anything. I remember to like using ``mkString`` in Scala very much. It was an easy to remember name, easy to use and very flexible to make Strings from collections.